### DISCUSSION TITLE
Installing Labgrid on Raspberry PI (cryptography compile problem)?

### DISCUSSION THREAD

**Discussion by Rentek8** [labgrid-project/labgrid#1190] [Q&A]
Hey Labgrid Team,

I'm working at a company and we want to integrate Labgrid into our Testing Environment. I tried in march to install it and had some issues with the installing part. Sometimes it works and sometimes not. We are using docker container and the python versions and everything else should be the same. I put some infos here, maybe some other have the same problems. It's not always the same problem. The python versions can't be the newest available. It is fixed. Also, the new installing from labgrid from april did not work as well.

![image](https://github.com/labgrid-project/labgrid/assets/134692075/38ebf3dc-8339-47d6-83db-e2acf21e50bb)

i installed rustup after that and the installation didn't work again.

![image](https://github.com/labgrid-project/labgrid/assets/134692075/bcd2cfc1-2008-44fd-98ff-c9582bb035a8)

Sometimes it is another python library. I follow the install guide from labgrid, but it has different problems on other raspberry pi's.

How do you test your current versions on ubuntu systems? Do you test it on Raspbian?

Info System:
![image](https://github.com/labgrid-project/labgrid/assets/134692075/c99e6c0c-dc4d-4931-a302-bfc97053f704)

Maybe it's the wrong project and i need to ask the question on the github page of the cryptography library. You can point me there, if the issue is wrong here.

Greetings
Niklas Kettner

**Comment by Emantor** [comment]
This is probably more suited to the `python-cryptography` package, but I recently saw a discussion in our matrix room/IRC channel where https://www.piwheels.org/ was mentioned as a solution to the problem.

**Reply by JulianPeD** [reply]
Hi,
I'm working together with Niklas on the same issue.
Right now we got another issue while installing the Labgrid framework on our raspberries.
Used Python version 3.9.13
At the moment, we only like to use the reservation feature of the labgrid framework. (USB-SD-Mux already in use)
Can you please tell us, which Python and/or Rasbian version is supported or recommended?
Regards
Julian Dross

**Reply by Bastian-Krause** [reply]
Supported Python versions are described in the `pyproject.toml`:

https://github.com/labgrid-project/labgrid/blob/9c2ac11b253544cf7e996ed492f393fcc9e2f8d3/pyproject.toml#L19

We test against Python 3.8, 3.9, 3.10 and 3.11 in [CI](https://github.com/labgrid-project/labgrid/actions/workflows/scheduled-unit-tests.yml).

The Raspbian/Raspberry Pi OS version should not really matter as long as you're using a virtualenv and update setuptools and pip in the virtualenv beforehand. Please desribe exactly what you're doing and show us the issue (text logs please, no screenshots).

**Reply by Shivar90** [reply]
Hi @Bastian-Krause ,
I try to install labgrid on Raspberry Pi 4 Model B .I take reference from labgrid document for installations "https://labgrid.readthedocs.io/en/latest/" . 
I am also facing same issue of " ERROR: Failed building wheel for cryptography
Successfully built labgrid
Failed to build cryptography".
Environment Setup: Please check attached images 
Python3 Version: Python 3.10.7
Please check attached logs.
Previously able to install labgrid(Coordinator, Client, Exporter on ubuntu 22.04) .Facing issue on Raspberry pi only. Please guide .

<img width="560" alt="Raspberrypi Os version" src="https://github.com/labgrid-project/labgrid/assets/36196330/29a60505-d1b3-47c7-808b-d25d702cec97">

<img width="941" alt="Rasperypi version" src="https://github.com/labgrid-project/labgrid/assets/36196330/cecc10b5-37b2-4157-bff6-3099944231b6">
[Labgrid-execution_logs.txt](https://github.com/labgrid-project/labgrid/files/12832933/Labgrid-execution_logs.txt)

**Reply by Bastian-Krause** [reply]
As mentioned above, you could try with piwheels.org:
```
pip3 install cryptography --index-url https://www.piwheels.org/simple
```
